src/terrain.cpp

```diff
@@ -199,7 +199,7 @@ terrain_type::terrain_type(const terrain_type& base, const terrain_type& overlay
 	editor_image_(base.editor_image_ + "~BLIT(" + overlay.editor_image_ +")"),
 	id_(base.id_+"^"+overlay.id_),
 	name_(overlay.name_),
-	editor_name_(base.editor_name_ + " / " + overlay.editor_name_),
+	editor_name_(base.editor_name_ + " / " + (overlay.editor_name_.empty() ? overlay.name_ : overlay.editor_name_)),	// Resolve Bug #23023: fall back to overlay name if overlay editor name is empty
```

Comment:
I'm all for writing code that speaks for itself and keeping the use of comments to only explain unusual things or explain intentions rather than restating code. I suppose in the late hour that I did this last night, I just wanted to make it clear what the story behind the bug was. Also, I find the ternary if operator less easy to read than the usual if-else block.

I did a check of editor_name_ and these seem to be the only instances where it's used. So I applied the fall back to name_ for base as well. I did a check of terrain entries without editor_name in terrain.cfg and the dozen or so that I found do seem to be mostly overlay-type terrains. So yes, probably only a potential problem for UMC, but good to fix as well.